pkg/controller/stack/stack_controller.go

```diff
@@ -138,16 +144,10 @@ func add(mgr manager.Manager, r *ReconcileStack) error {
 		return err
 	}
 
-	// Filter out update events if an object's metadata.generation is
-	// unchanged, or if the object never had a generation update.
-	//  - https://github.com/operator-framework/operator-lib/blob/main/predicate/nogeneration.go#L29-L34
-	//  - https://github.com/operator-framework/operator-sdk/issues/2795
-	//  - https://github.com/kubernetes-sigs/kubebuilder/issues/1103
-	//  - https://github.com/kubernetes-sigs/controller-runtime/pull/553
-	//  - https://book-v1.book.kubebuilder.io/basics/status_subresource.html
-	// Set up predicates.
+	// Filter for update events where an object's metadata.generation is changed (no spec change!),
+	// or the "force reconcile" annotation is used (and not marked as handled).
 	predicates := []predicate.Predicate{
-		predicate.Or(predicate.GenerationChangedPredicate{}, libpredicate.NoGenerationPredicate{}),
+		predicate.Or(predicate.GenerationChangedPredicate{}, ReconcileRequestedPredicate{}),
```

Comment:
Predicates give us finer control of the Kubernetes events the controller sees, and allows it to control which resources need to be reconciled.